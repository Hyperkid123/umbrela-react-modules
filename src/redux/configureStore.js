import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
