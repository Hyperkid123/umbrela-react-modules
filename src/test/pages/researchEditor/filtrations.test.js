import React from 'react'
import Filtrations, {Filtrations as Snapshot} from '../../../pages/researchEditorModule/filtrations';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('Filtrations component empty', () => {
  it('should render correclty' , () => {
      const tree = shallow(
        <Snapshot getSheets={jest.fn()}/>
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });
  const mockStore = configureStore();
  let initialState = {
    editor: {
      sheets: [],
      activeSheet: {}
    }
  };
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<Filtrations store={store}/>);
    expect(wrapper).toBeDefined();
  })

  it('should render correclty with active sheet' , () => {
      const tree = shallow(
        <Snapshot getSheets={jest.fn()} activeSheet/>
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });
});
