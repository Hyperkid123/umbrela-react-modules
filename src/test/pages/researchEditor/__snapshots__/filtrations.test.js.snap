// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filtrations component should render correclty 1`] = `
<DNDContext>
  <WithStyles
    item={true}
    lg={4}
    md={4}
    sm={12}
    xl={3}
    xs={12}
  >
    <Connect(SheetMenu)
      hideNewSheet={true}
    />
  </WithStyles>
  <WithStyles
    item={true}
    lg={8}
    md={8}
    sm={12}
    xl={9}
    xs={12}
  >
    <WithStyles
      container={true}
      spacing={0}
    />
  </WithStyles>
</DNDContext>
`;
